lasagne/updates.py

```diff
@@ -262,10 +251,52 @@ def nesterov_momentum(loss_or_grads, params, learning_rate, momentum=0.9):
 
 
 def adagrad(loss_or_grads, params, learning_rate=1.0, epsilon=1e-6):
+    """Adagrad updates [1]
```

Comment:
From [the guide](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#sections) (down at "11. References"), I think the syntax should be `[1]_` to reference something, and `[1]` to define the reference. Haven't tried it myself, though.